core/types/access_list_tx.go

```diff
@@ -42,6 +42,24 @@ func (al AccessList) StorageKeys() int {
 	return sum
 }
 
+// Equal returns true if two AccessLists are equal.
```

Comment:
Nitpick, but maybe add a small comment that the same content in a different order is considered different. (since semantically, gas cost / EVM execution wise the order doesn't matter).